software/main.py

```diff
@@ -0,0 +1,97 @@
+from ast import Break
+from asyncio import constants
+from concurrent.futures import process
+from pickle import TRUE
+import image_processor
+import camera
+#import motion
+import cv2
+import time
+from math import *
+import numpy as np
+import movement
+
+spin = 10
+
+
+
+
+
+def main_loop():
+    debug=True
+    
+    cam=camera.RealsenseCamera(exposure=100)
+    processor = image_processor.ImageProcessor(cam, debug=debug, color_config = "colors/colors.pkl")
+    global robot
+    robot=movement.OmniRobot()
+    
+    processor.start()
+    start=time.time()
+    fps=0
+    frame=0
+    frame_cnt=0
+    
+    try:
+        while True:
+            processedData= processor.process_frame(aligned_depth=False)
+            frame_cnt+=1
+            frame+=1
+            #robot.try_motors()
+            #break
+            if frame%30==0:
+                frame = 0
+                end = time.time()
+                fps = 30/(end-start)
+                start = end
+                #[Object: x=7; y=212; size=15.0; distance=212; exists=True]  
+                print(processedData.balls)  
+        
+                if (len(processedData.balls)!=0):
+                    processedData.balls.sort(key= lambda x: x.size)
+                    xcord=processedData.balls[-1].x
+                    ycord=processedData.balls[-1].y
+                    dist=processedData.balls[-1].distance
+
+                    if dist > 400:
+                        robot.center_ball(xcord)
+                    else:
+                        #robot.center_ball(spin, xcord)
+                        wheel1=robot.omni(xcord,1,dist)

```

Comment:
Check https://ut-robotics.github.io/picr22-home/basketball_robot_guide/software/review_summary.html#_do_not_duplicate_code

Also, you should calculate motion planning in a separate module or in the main. The robots electronics do not know anything about x and y coordinates and logically should just be provided the goal speeds. This solution also locks you into a motion curve that can not be tweaked on a state-to-state basis.